app/Http/Controllers/ClassroomController.php

```diff
@@ -4,79 +4,71 @@
         } else {
-            return response()->json(['message' => 'Class not found'], );
+            return $this->jsonResponse(['message' => 'Class not found'], 404);
         }
     }
 
     // Insert a new row into the 'classes' table
     public function create(Request $request)
     {
+        // Validate input
         $request->validate([
-            'name' => 'required|string|max:255',
-            'description' => 'nullable|string',
-            // Add other validation rules as necessary
+            'title' => 'required',
+            'subject' => 'required|in:Sains,Matematika,Bahasa,Teknologi,Sosial,Seni',
         ]);
 
-        $class = new Classes();
-        $class->name = $request->input('name');
-        $class->description = $request->input('description');
-        // Set other fields as necessary
-
-        $class->save();
+        Classes::create([
+            'id' => Str::uuid(),
+            'title' => $request->input('title'),
+            'description' => $request->input('description'),
+            'subject' => $request->input('subject'),
+            'photo' => $request->input('photo'),
+        ]);
 
-        return response()->json($class, 201);
+        return $this->jsonResponse(['message' => 'Class created successfully'], 201);
     }
 
     // Update a row by ID in the 'classes' table
     public function update(Request $request, $id)
     {
+        // Validate input
         $request->validate([
-            'name' => 'required|string|max:255',
-            'description' => 'nullable|string',
-            // Add other validation rules as necessary
+            'title' => 'required',
+            'subject' => 'required|in:Sains,Matematika,Bahasa,Teknologi,Sosial,Seni',
         ]);
 
-        $class = Classes::find($id);
-        if ($class) {
-            $class->name = $request->input('name');
-            $class->description = $request->input('description');
-            // Update other fields as necessary
-
-            $class->save();
+        Classes::where('id', $id)->update([
+            'title' => $request->input('title'),
+            'description' => $request->input('description'),
+            'subject' => $request->input('subject'),
+            'photo' => $request->input('photo'),
+        ]);
 
-            return response()->json($class);
-        } else {
-            return response()->json(['message' => 'Class not found'], 404);
-        }
+        return $this->jsonResponse(['message' => 'Class updated successfully']);
     }
 
     // Delete a row by ID in the 'classes' table
     public function delete(Request $request, $id)
     {
-        $class = Classes::find($id);
-        if ($class) {
-            $class->delete();
-            return response()->json(['message' => 'Class deleted successfully']);
-        } else {
-            return response()->json(['message' => 'Class not found'], 404);
-        }
+        Classes::where('id', $id)->delete();
```

Comment:
**suggestion (bug_risk):** Add existence check before deleting

Similar to the update method, the delete method doesn't check if the class exists before attempting to delete. This could lead to silent failures. Consider adding an existence check and returning an appropriate error response if the class is not found.